wyzesense2mqtt/wyzesense.py

```diff
@@ -303,23 +352,80 @@ def _OnEventLog(self, pkt):
         # assert msg_len + 8 == len(pkt.Payload)
         tm = datetime.datetime.fromtimestamp(ts / 1000.0)
         msg = pkt.Payload[9:]
-        log.info("LOG: time=%s, data=%s", tm.isoformat(), bytes_to_hex(msg))
+        log.info(f"LOG: time={tm.isoformat()}, data={bytes_to_hex(msg)}")
 
-    def __init__(self, device, event_handler):
-        self.__lock = threading.Lock()
-        self.__fd = os.open(device, os.O_RDWR | os.O_NONBLOCK)
-        self.__sensors = {}
-        self.__exit_event = threading.Event()
-        self.__thread = threading.Thread(target=self._Worker)
-        self.__on_event = event_handler
+    def _OnHMSEvent(self, pkt):
+        _, sensor_mac = struct.unpack_from(">B8s", pkt.Payload)
+        sensor_mac = sensor_mac.decode('ascii')
+        timestamp = datetime.datetime.utcnow()
 
-        self.__handlers = {
-            Packet.NOITFY_SYNC_TIME: self._OnSyncTime,
-            Packet.NOTIFY_SENSOR_ALARM: self._OnSensorAlarm,
-            Packet.NOTIFY_EVENT_LOG: self._OnEventLog,
+        event_data = pkt.Payload[10:]
+        event_type = event_data[4]
+        battery = int(event_data[2] / 155 * 100)
```

Comment:
I'm really guessing here on the battery level for the keypad...

The max value I could get from it was `155`, with brand new batteries, and swapping out the oldest batteries I had laying around gave me a minimum value of `142`, so I don't know that this calculation is quite right, but it's still open to change.